migrate/starter/starter.go

```diff
@@ -61,6 +62,11 @@ func checkStartVersion(args []string) version {
 	if err != nil {
 		return internalV2
 	}
+	if fs.Lookup("version").Value.String() == "true" {
+		fmt.Println("etcd version:", etcdversion.Version, "(allow legacy mode)")
```

Comment:
@crawford It is a little different between these two. When running etcd v2.0, it autoupgrades the data dir and runs it in 2.0 way. While running etcd v2.0 (allow legacy mode), it syscall.Exec v0.4 binary to run v0.4 data dir in old way if not upgraded. It doesn't change the underlying data.